src/rgw/rgw_http_client.h

```diff
@@ -102,6 +102,52 @@ class RGWHTTPClient
   int get_req_retcode();
 };
 
+
+class RGWHTTPHeadersCollector : public RGWHTTPClient {
+public:
+  /* Case insensitive comparator for containers carrying HTTP headers. */
+  struct CILess : public std::binary_function<std::string, std::string, bool> {
+    bool operator()(const std::string& lhs,
+                    const std::string& rhs) const {
+      return ::strcasecmp(lhs.c_str(), rhs.c_str()) < 0 ;
+    }
+  };
+
+  typedef std::string header_name_t;
+  typedef std::string header_value_t;
+  typedef std::set<header_name_t, CILess> header_spec_t;
```

Comment:
I picked up this name to reflect the intention that the type is solely for passing information about in which headers caller is interested. Using `std::set` is only a private implementation detail.